src/ContaoCommunityAlliance/DcGeneral/Contao/View/Contao2BackendView/Widget/FileTree.php

```diff
@@ -0,0 +1,346 @@
+        }
+
+        if ($downloads) {
+            if ($this->isAllowedDownload($file->extension)) {
+                return \Image::getHtml($file->icon) . ' ' . $info;
+            }
+
+            return false;
+        }
+
+        if (!$file->isGdImage && !$file->isSvgImage) {
+            return \Image::getHtml($file->icon) . ' ' . $info;
+        }
+
+        $image = 'placeholder.png';
+
+        if ($file->isSvgImage
+            || $file->height <= \Config::get('gdMaxImgHeight') && $file->width <= \Config::get('gdMaxImgWidth')
+        ) {
+            $image = \Image::get($model->path, 80, 60, 'center_center');
+        }
+
+        return \Image::getHtml($image, '', 'class="gimage" title="' . specialchars($info) . '"');
+    }
+
+    /**
+     * Apply the sorting to the icons.
+     *
+     * @param array $icons The file icons.
+     *
+     * @return array
+     */
+    private function applySorting($icons)
+    {
+        if ($this->orderField != '' && is_array($this->orderFieldValue)) {
+            $ordered = array();
+
+            foreach ($this->orderFieldValue as $uuid) {
+                if (isset($icons[$uuid])) {
+                    $ordered[$uuid] = $icons[$uuid];
+                    unset($icons[$uuid]);
+                }
+            }
+
+            foreach ($icons as $uuid => $icon) {
+                $ordered[$uuid] = $icon;
+            }
+
+            $icons = $ordered;
+        }
+
+        return $icons;
+    }
+
+    /**
+     * Generate the adjust selection link.
+     *
+     * @param array $values The selected files.
+     *
+     * @return string
+     */
+    private function generateLink($values)
+    {
+        $inputProvider = $this->getEnvironment()->getInputProvider();
+
+        // Contao passed File ids sinc 3.3.4
+        // @see https://github.com/contao/core/commit/c1472209fdfd6e2446013430753ed65530b5a1d1
+        if (version_compare(VERSION, '3.3.4', '>=')) {
+            $values = array_keys($values);
+        } else {
+            $values = array_map('String::binToUuid', $values);
+        }
+
+        return sprintf(
+            'contao/file.php?do=%s&amp;table=%s&amp;field=%s&amp;act=show&amp;id=%s&amp;value=%s&amp;rt=%s',
+            $inputProvider->getParameter('do'),
+            $this->getModel()->getProviderName(),
+            $this->strField,
+            $this->getModel()->getId(),
+            implode(',', $values),
+            \RequestToken::get()
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function generate()
+    {
+        $values = array();
+        $icons  = array();
+
+        if (!empty($this->varValue)) {
+            $files = \FilesModel::findMultipleByUuids((array)$this->varValue);
+            $this->renderList($values, $icons, $files, true);
+            $icons = $this->applySorting($icons);
+        }
+
+        $translator = $this->getEnvironment()->getTranslator();
+        $template   = new \BackendTemplate($this->subTemplate);
```

Comment:
Please use `ContaoCommunityAlliance\DcGeneral\Contao\View\Contao2BackendView\ContaoBackendViewTemplate` here, which implements our interface and provides setter and getter functions.
We also could add the translate function to that template class and interface as well.